### PR TITLE
Feat workflow collector

### PR DESCRIPTION
**Pull request by bnjjj**
Please CHECK SQL MIGRATION NUMBER before merge :)

### CODE REVIEW

File: engine/api/worker/worker.go [hunk 1]
```diff
@@ -44,27 +45,45 @@ func DeleteWorker(db *gorp.DbMap, id string) error {
...
+
+		switch pbJobType.String {
+		case sdk.JobTypePipeline:
+			if err := pipeline.RestartPipelineBuildJob(tx, jobID.Int64); err != nil {
+				log.Error("DeleteWorker[%d]> Cannot restart pipeline build job: %s", id, err)
+			} else {
+				log.Info("DeleteWorker[%d]> PipelineBuildJob %d restarted after crash", id, jobID.Int64)
+			}
+		case sdk.JobTypeWorkflowNode:
+			wNodeJob, errL := workflow.LoadNodeJobRunWithoutStore(tx, jobID.Int64)
```
Comment (fsamin) [inline engine/api/worker/worker.go]: why not modify LoadNodeJobRun to handle nil store ?

File: engine/api/workflow/dao_node_run_job.go [hunk 2]
```diff
@@ -101,6 +101,17 @@ func LoadNodeJobRun(db gorp.SqlExecutor, store cache.Store, id int64) (*sdk.Work
 	return &job, nil
 }
 
+//LoadNodeJobRunWithoutStore load a NodeJobRun given its ID without a given store
+func LoadNodeJobRunWithoutStore(db gorp.SqlExecutor, id int64) (*sdk.WorkflowNodeJobRun, error) {
```
Comment (fsamin) [inline engine/api/workflow/dao_node_run_job.go]: i think it better to only keep LoadNodeJobRun

File: engine/sql/061_pipeline_build_job_retry.sql [hunk 3]
```diff
@@ -0,0 +1,6 @@
+-- +migrate Up
+ALTER TABLE workflow_node_run_job ADD COLUMN retry int DEFAULT 0;
+ALTER TABLE worker ADD COLUMN job_type text;
```
Comment (fsamin) [inline engine/sql/061_pipeline_build_job_retry.sql]: should we update workflow_node_run_job to set the default value on existing tuples ?

Comment (bnjjj) [reply engine/sql/061_pipeline_build_job_retry.sql]: Yes we can !

**Comment by bnjjj** [comment]
In order to prod you need to stop hatchery, wait all workers are gone and prod

File: engine/sql/061_pipeline_build_job_retry.sql (hunk 3 above)
Comment (bnjjj) [reply engine/sql/061_pipeline_build_job_retry.sql]: No, see PR comment below

**Comment by yesnault** [comment]
rebased master & resolving conflicts done

**Comment by yesnault** [comment]
Ready to merge, run 629. UT / IT OK

**Pull request merged**